From 2beb82c292725ad500db9d564ca73dd6c2bce463 Mon Sep 17 00:00:00 2001
From: Julien Cristau <jcristau@debian.org>
Date: Sun, 23 Aug 2009 12:31:21 +0200
Subject: [PATCH] pixman/arm: don't use env vars to get hwcap/platform

---
 pixman/pixman-cpu.c |    6 ++++++
 1 files changed, 6 insertions(+), 0 deletions(-)

diff --git a/pixman/pixman-cpu.c b/pixman/pixman-cpu.c
index a2a7b8a..af16715 100644
--- a/pixman/pixman-cpu.c
+++ b/pixman/pixman-cpu.c
@@ -253,8 +253,10 @@ pixman_arm_read_auxv ()
 	    if (aux.a_type == AT_HWCAP)
 	    {
 		uint32_t hwcap = aux.a_un.a_val;
+#if 0
 		if (getenv ("ARM_FORCE_HWCAP"))
 		    hwcap = strtoul (getenv ("ARM_FORCE_HWCAP"), NULL, 0);
+#endif
 		/* hardcode these values to avoid depending on specific
 		 * versions of the hwcap header, e.g. HWCAP_NEON
 		 */
@@ -266,8 +268,10 @@ pixman_arm_read_auxv ()
 	    else if (aux.a_type == AT_PLATFORM)
 	    {
 		const char *plat = (const char*) aux.a_un.a_val;
+#if 0
 		if (getenv ("ARM_FORCE_PLATFORM"))
 		    plat = getenv ("ARM_FORCE_PLATFORM");
+#endif
 		if (strncmp (plat, "v7l", 3) == 0)
 		{
 		    arm_has_v7 = TRUE;
@@ -281,11 +285,13 @@ pixman_arm_read_auxv ()
 	}
 	close (fd);
 
+#if 0
 	/* if we don't have 2.6.29, we have to do this hack; set
 	 * the env var to trust HWCAP.
 	 */
 	if (!getenv ("ARM_TRUST_HWCAP") && arm_has_v7)
 	    arm_has_neon = TRUE;
+#endif
     }
 
     arm_tests_initialized = TRUE;
-- 
1.6.3.3

